### ISSUE TITLE
Interaction not working properly when multiple stacking contexts exist.

### ISSUE THREAD

**Issue by KevinMoran99** [usablica/intro.js#1202]
## Description
I'm working on a project in which I'm trying to implement introjs, but I've noted that when highlighting an element that's inside of a different stacking context (i.e. an input inside a modal with z-index: 1000), the element does get highlited properly, but interaction in the element is not posible, even if disableInteraction: false is explicitly set in the option's configuration. 

## Expected Behavior
Interaction should be allowed on highlited elements inside another stacking context.

## Actual Behavior
Interaction does not work in this case, clicking the highlited element does nothing.

## Errors and Screenshots 
When right click -> inspecting the highlited box, dev tools takes me to the div with "introjs-helperLayer" class, oposed to normal behaviour in which inspecting the box takes me to the actual element. For what I have seen, this happens because, even when the element gets the "showElement" class that gives it an enormous z-index, it is still in the modal's stacking context, so it cannot be put over the helperLayer which exists in the root stacking context. I've read through the docs to see if there's an existing solution to this problem, but I haven't found anything (don't know if there's actually a common solution that I have overlooked).
![image](https://user-images.githubusercontent.com/35577671/113769617-6bc65c00-96de-11eb-949e-e8dc1eee68c6.png)
![image](https://user-images.githubusercontent.com/35577671/113769820-a3350880-96de-11eb-9695-e2f74a14eb25.png)

## Environment (optional)
Tried on both Chrome and Firefox, introjs version is 3.4.0, we're developing an Angular application using PrimeNG.

**Comment by ajjones20** [comment]
I am having this same issue. I want interactivity in a bootstrap modal, rearranging the DOM doesn't seem to resolve the problem either.

**Comment by afshinm** [comment]
I can look into this but I believe this is an exception where element interaction doesn't work because you have an absolute/fixed parent. 

Let me know if anyone has any suggestions.

**Comment by sdraper69** [comment]
I too get this issue, I agree it seems to be because the target's parent is absolutely positioned.

Unfortunately this makes intro unusable for me. It sounds like there is no workaround? 

If in chrome dev tools I take untick the position: absolute then I can see its actually selectable, to me this would suggest that however you are creating the "hole" to click it just isn't looking at the elements actual position.

**Comment by KevinMoran99** [comment]
> I can look into this but I believe this is an exception where element interaction doesn't work because you have an absolute/fixed parent.
> 
> Let me know if anyone has any suggestions.

Indeed, the parent elements of the components that give me this issue are either absolute or fixed. One kinda hacky solution that I found was to analyze the structure of said components and identify specifically which elements of the DOM generated new stacking contexts.

So, I declared the below functions. 
The first one gets called when I initialize the intro.js tour, and gets rid of the modal's stacking context.
The second one then gets called in .onbeforeexit(), and makes sure that the modal recovers it's original properties.
![image](https://user-images.githubusercontent.com/35577671/116169779-baa25880-a6c2-11eb-960d-cbffd2ddebc3.png)

That way, I managed to have everything that I needed in the same stacking context, and hence could use the interaction inside the modal. However, this 'solution' it's still really problematic, as I have to specifically analyze any component that generates a stacking context. There are still some bugs that happen when I do this that I haven't solved yet, and I believe that those could be a consequence of playing this way with the DOM's properties (for example, when I highlight a button that's inside the footer of the modal, the intro.js overlay scrolls down in order to move that button to the middle of the viewport, generating an unusual scroll on my application).

I don't know if this approach, as buggy and problematic as it may seem, could help to conceive an idea to properly fix this, haha. Another idea that went through my head was to somehow make the intro.js helpers to render inside the target's stacking context, but as I had no idea on how to make that, I decided to go for the approach of removing the modal's stacking context and then restoring it.

**Comment by ajjones20** [comment]
this solution works for me as well. thank you so much

**Comment by xiaocaohulian** [comment]
![微信图片_20210507150608](https://user-images.githubusercontent.com/37647755/117411852-a3772c80-af46-11eb-8a98-2014229b11c2.png)
![微信图片_20210507151035](https://user-images.githubusercontent.com/37647755/117411866-a83be080-af46-11eb-9b06-86b8f101a5a2.png)
![1620371535(1)](https://user-images.githubusercontent.com/37647755/117411890-af62ee80-af46-11eb-9710-55830e4cbdc9.png)
如果目标元素的父元素有z-index,此时遮罩层会完全覆盖住,将遮罩层两个元素移动到目标元素同级中,可以避免覆盖
If the parent element of the target element has a z-index, then the mask will be completely overwritten. Moving the two mask elements to the same level as the target element can avoid overwriting

**Comment by afshinm** [comment]
fyi I'm still working on this and hoping to provide a solution soon. I added a couple of Cypress tests to make sure we can catch this.

**Comment by afshinm** [comment]
To summarize this thread: You cannot interact with the highlighted element if the target element's parent has the `z-index` value set to anything other than `auto` (i.e stacking contexts). For now, I'm going to focus on adding more tests to capture the current state.

@xiaocaohulian I like your suggestion and I believe can fix the issue. Can you send a PR?

**Comment by rynkk** [comment]
Has there been any progress on this issue? 
Currently I'm stuck on some ancient version of intro.js because this problem did not exist at that point.

**Comment by afshinm** [comment]
@rynkk not yet unfortunately. what version of intro.js are you using right now?

**Comment by rynkk** [comment]
@afshinm 
I am using 2.9.3.
I'm gonna try to pin point were the regression happened.

**Comment by chenxu-javascript** [comment]
@xiaocaohulian
![image](https://user-images.githubusercontent.com/16332458/124581745-f5272f00-de83-11eb-86e5-d2f67408eeac.png)
![image](https://user-images.githubusercontent.com/16332458/124581803-07a16880-de84-11eb-8c85-a0d9d1c1c51a.png)

**Comment by scifresh** [comment]
Any progress on this issue?
I have the exact same problems with v5.0.0.

**Comment by justhzd** [comment]
是否可以把遮罩层修改为使用svg实现，因为svg是可以镂空的，就不会遮挡交互了。
Whether the mask layer can be modified to use SVG implementation, because SVG can be hollowed out, so it will not block the interaction.

**Comment by justhzd** [comment]
只需要在创建introJs对象的时候，设置选择器到带z-index的父级元素里就可以了
比如有一个modal。<div id="add-modal" class="modal fade" tabindex="-1" role="dialog" aria-hidden="true">
那初始化introJs时这样用
$('#add-modal').on('shown.bs.modal', function () {
introJs('#add-modal').setOptions({
                                disableInteraction: false,
                                steps: []}).start();
});


Just set the selector to the parent element with Z-index when creating the introjs object
For example, there is a modal< div id="add-modal" class="modal fade" tabindex="-1" role="dialog" aria-hidden="true">
This is used when initializing introjs
$('#add-modal'). on('shown.bs.modal', function () {
introJs('#add-modal'). setOptions({
disableInteraction: false,
steps: []}). start();
});

**Comment by justhzd** [comment]
@scifresh I solved it. Check my message and hope it can help you.

**Comment by justhzd** [comment]
@scifresh I found a solution. Check my message and hope it can help you.

**Comment by justhzd** [comment]
@scifresh use introJs([targetElm])

**Comment by jiangbinwangyi** [comment]
.introjs-overlay {
  pointer-events: none;
}
.introjs-helperLayer {
  pointer-events: none;
}

[https://developer.mozilla.org/zh-CN/docs/Web/CSS/pointer-events](pointer-events - CSS：层叠样式表 | MDN)

**Comment by bluestarstudios** [comment]
The above CSS option is great. Since as of version 7.2.0, this still hasn't been fixed, here's an alternate solution, to dynamically make the elements clickable or not.
First, create this CSS rule:
```
.introjs-overlay.no-pointer-events, .introjs-helperLayer.no-pointer-events {
    pointer-events: none;
}
```
Then you can dynamically make the elements "clickable" by applying the new no-pointer-events classes (here I'm doing it for step number 2):
```
introJs().onafterchange(function (targetElement) {
    jQuery('.introjs-overlay, .introjs-helperLayer').removeClass('no-pointer-events');
    if (introJs()._currentStep === 2) {
        jQuery('.introjs-overlay, .introjs-helperLayer').addClass('no-pointer-events');
    }
});
```
